logging/logging-processor/src/main/java/org/infinispan/logging/processor/InfinispanLoggingProcessor.java

```diff
@@ -0,0 +1,127 @@
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import javax.tools.StandardLocation;
+
+import org.infinispan.logging.processor.report.XmlReportWriter;
+import org.jboss.logging.processor.apt.ProcessingException;
+import org.kohsuke.MetaInfServices;
+/**
+ * InfinispanLoggingProcessor.
+ * @author Durgesh Anaokar
+ * @Since 13.0
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedAnnotationTypes({InfinispanLoggingProcessor.DESCRIPTION})
+@MetaInfServices(Processor.class)
+public class InfinispanLoggingProcessor extends AbstractProcessor {
+   static final String DESCRIPTION = "org.infinispan.logging.annotations.Description";
+   static final String MESSAGE = "org.jboss.logging.annotations.Message";
+   static final String LOGMESSAGE = "org.jboss.logging.annotations.LogMessage";
+
+   @Override
+   public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+      for (TypeElement annotation : annotations) {
+         try {
+            createReportXml(roundEnv, annotation);
+         } catch (Throwable t) {
+            uncaughtException(t);
+            t.printStackTrace();
+            return false;
+         }
+      }
+      return true;
+   }
+
+   private void createReportXml(RoundEnvironment roundEnv, TypeElement annotation) throws IOException {
+      try {
+         Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
+         Map<String, List<Element>> mapWithQualifiedName = getMapWithQualifiedName(annotatedElements);
+         for (Map.Entry<String, List<Element>> entry : mapWithQualifiedName.entrySet()) {
+
+            String qualifiedName = entry.getKey();
+            final int lastDot = qualifiedName.lastIndexOf(".");
+            String packageName = qualifiedName.substring(0, lastDot);
+            String simpleName = qualifiedName.substring(lastDot + 1);
+            BufferedWriter bufferedWriter = createWriter(packageName, simpleName + ".xml");
+            XmlReportWriter reportWriter = new XmlReportWriter(bufferedWriter);
+            reportWriter.writeHeader(qualifiedName);
+            List<Element> list = entry.getValue();
+            for (Element element : list) {
+               try {
+                  reportWriter.writeDetail(element);
+               } catch (ProcessingException e) {
+                  final AnnotationMirror a = e.getAnnotation();
+                  final AnnotationValue value = e.getAnnotationValue();
+                  error(element, "Error while processing the report: for annotation" + a.toString() + " for value: "
+                        + value.toString(), e);
+               }
+            }
+            reportWriter.writeFooter();
+            reportWriter.close();
+            bufferedWriter.close();
```

Comment:
If you create the `bufferedWriter` and `reportWriter` in a try-with-resources` statement, you don't have to explicitly call `.close()`.